### PR TITLE
Potions Extractor

### PR DESCRIPTION
**Pull request by SelfMadeSystem**
# Objective

We need a way to acquire (and thus extract) all potion/status effects + their attributes

Also, #558 was getting really large.

# Solution

I made an extractor.

Also, as per [the request](https://github.com/valence-rs/valence/pull/558#issuecomment-1760283675) of @dyc3, I split #558's extractor code into this pr.

### CODE REVIEW

File: crates/valence_generated/build/status_effects.rs [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+            quote! {
+                Self::#name => #id,
+            }
+        })
+        .collect::<TokenStream>();
+
+    let effect_from_ident_arms = effects
+        .iter()
+        .map(|effect| {
+            // TODO: put the full resource identifier in the extracted JSON.
```
Comment (dyc3) [inline crates/valence_generated/build/status_effects.rs:59]: Does this TODO need to be resolved in this PR?

**Review by dyc3 — COMMENTED** [review]
Looks good.

File: crates/valence_generated/build/status_effects.rs (hunk 1 above)
Comment (SelfMadeSystem) [reply crates/valence_generated/build/status_effects.rs:59]: I copied the TODO from `sound.rs`

Comment (SelfMadeSystem) [reply crates/valence_generated/build/status_effects.rs:59]: I would say no

Comment (dyc3) [reply crates/valence_generated/build/status_effects.rs:59]: Ok, then lets remove that comment.

Comment (SelfMadeSystem) [reply crates/valence_generated/build/status_effects.rs]: > Ok, then lets remove that comment.

alright

**Review by dyc3 — APPROVED** [review]

**Pull request merged**